docs/contributing.rst

```diff
@@ -81,47 +81,70 @@ Thank you for your help!
+and ``libevmone.dylib`` on macOS.
+
+Running the tests
+-----------------
 
 Solidity includes different types of tests, most of them bundled into the
 `Boost C++ Test Framework <https://www.boost.org/doc/libs/1_69_0/libs/test/doc/html/index.html>`_ application ``soltest``.
 Running ``build/test/soltest`` or its wrapper ``scripts/soltest.sh`` is sufficient for most changes.
 
-Some tests require the ``evmone`` library, others require ``libz3``.
+The ``./scripts/tests.sh`` script executes most Solidity tests automatically,
+including those bundled into the `Boost C++ Test Framework <https://www.boost.org/doc/libs/1_69_0/libs/test/doc/html/index.html>`_ application ``soltest`` (or its wrapper ``scripts/soltest.sh``),
+as well as command line tests and compilation tests.
 
-The test system will automatically try to discover the location of the ``evmone`` library
+The test system automatically tries try to discover the location of the ``evmone`` library
 starting from the current directory. The required file is called ``libevmone.so`` on Linux systems,
-``evmone.dll`` on Windows systems and ``libevmone.dylib`` on MacOS. If it is not found, the relevant tests
-are skipped. To run all tests, download the library from
-`Github <https://github.com/ethereum/evmone/releases/tag/v0.3.0>`_
-and either place it in the project root path or inside the ``deps`` folder.
-
-If you do not have libz3 installed on your system, you should disable the SMT tests:
-``./scripts/soltest.sh --no-smt``.
+``evmone.dll`` on Windows systems and ``libevmone.dylib`` on macOS. If it is not found, tests that
+use it are skipped. These tests are ``libsolididty/semanticTests``, ``libsolidity/GasCosts``,
+``libsolidity/SolidityEndToEndTest``, part of the soltest suite. To run all tests, download the library from
+`GitHub <https://github.com/ethereum/evmone/releases/tag/v0.3.0>`_
+and place it in the project root path or inside the ``deps`` folder.
 
-``./build/test/soltest --help`` has extensive help on all of the options available.
-See especially:
+If the ``libz3`` library is not installed on your system, you should disable the
+SMT tests using the ``--no-smt`` option when running ``./scripts/tests.sh`` or
+``./scripts/soltest.sh``. These tests are ``libsolidity/smtCheckerTests`` and ``libsolidity/smtCheckerTestsJSON``.
 
-- `show_progress (-p) <https://www.boost.org/doc/libs/1_69_0/libs/test/doc/html/boost_test/utf_reference/rt_param_reference/show_progress.html>`_ to show test completion,
-- `run_test (-t) <https://www.boost.org/doc/libs/1_69_0/libs/test/doc/html/boost_test/utf_reference/rt_param_reference/run_test.html>`_ to run specific tests cases, and
-- `report-level (-r) <https://www.boost.org/doc/libs/1_69_0/libs/test/doc/html/boost_test/utf_reference/rt_param_reference/report_level.html>`_ give a more detailed report.
+.. csv-table:: The full test suite
+   :file: ./contributing/tests-list.csv
```

Comment:
I don't think this is very helpful. What is the information we convey here? This list will be outdated very soon and I don't think we want to have the overhead of updating it.

Reply:
I have no knowledge of what tests are changing and when, and this PR has sat here for a long time. I'm reacting to the Zeppelin audit and comments from contributors that they don't understand what tests we have and what they do, hence a reference might be useful to them. Otherwise there are a lot of tests that are supposed to pass before someone can make a contribution, but most people (including some core team hires) take time to understand what they are. So a reference seems helpful to all of these people. If things change, that's fine, but a reference is still useful.

If you render the RsT, or look at the csv it lists what tests run as part of which test suite and what they are testing.

I am open to other suggestions (like maybe commenting the test files instead), but as this has come up a bit, I feel it's valuable to contributors.

Reply:
@chriseth opinion after @ChrisChinchilla 's reply?

Reply:
@bshastry The error you pointed out was actually a bug in the SMTChecker docs, should be fixed by https://github.com/ethereum/solidity/pull/8470. The `--no-smt` flag works for `soltest` and `isoltest`, but doesn't relay anything to the command line tests.

Reply:
@leonardoalt Are you referring to the comment about SMT checker not being disabled by the bash tests script (https://github.com/ethereum/solidity/pull/7632#discussion_r358373498) or the other one about string comparison in bash (https://github.com/ethereum/solidity/pull/7632#discussion_r358379555)?

Reply:
First one

Reply:
> The --no-smt flag works for soltest and isoltest, but doesn't relay anything to the command line tests.

I'm a bit confused. Do you mean `$ ./scripts/tests.sh --no-smt` will work as intended post #8470 ?

Reply:
`--no-smt` has always worked when relayed to `soltest` and `isoltest`. What didn't work was the command line tests when no solver is available. That is fixed by #8470

Reply:
Okay, thanks @leonardoalt 

In that case, please ignore my earlier comment

Reply:
@chriseth The `csv` file was removed